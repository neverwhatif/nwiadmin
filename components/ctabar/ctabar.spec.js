import React from 'react';
import { shallow } from 'enzyme';

import CTABar from './index';

jest.mock('nwiadmin/components/button', () => 'Button');

const data = [{
    label: 'Lorem',
    action: jest.fn(),
}, {
    label: 'Ipsum',
    action: jest.fn(),
}];

describe('CTABar', () => {
    const component = shallow(<CTABar data={data} />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
    });
    it('should render buttons', () => {
        expect(component.find('Button').length).toBe(2);
    });
    it('should render button content', () => {
        expect(component.find('Button').first().key()).toBe('Lorem');
    });
});
