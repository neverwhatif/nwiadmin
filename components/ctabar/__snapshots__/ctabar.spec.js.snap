// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CTABar should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CTABar
    data={
      Array [
        Object {
          "action": [MockFunction],
          "label": "Lorem",
        },
        Object {
          "action": [MockFunction],
          "label": "Ipsum",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          onClick={[MockFunction]}
        >
          Lorem
        </Button>,
        <Button
          onClick={[MockFunction]}
        >
          Ipsum
        </Button>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Lorem",
        "nodeType": "host",
        "props": Object {
          "buttonStyle": undefined,
          "children": "Lorem",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "Lorem",
        "type": "Button",
      },
      Object {
        "instance": null,
        "key": "Ipsum",
        "nodeType": "host",
        "props": Object {
          "buttonStyle": undefined,
          "children": "Ipsum",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "Ipsum",
        "type": "Button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            onClick={[MockFunction]}
          >
            Lorem
          </Button>,
          <Button
            onClick={[MockFunction]}
          >
            Ipsum
          </Button>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Lorem",
          "nodeType": "host",
          "props": Object {
            "buttonStyle": undefined,
            "children": "Lorem",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Lorem",
          "type": "Button",
        },
        Object {
          "instance": null,
          "key": "Ipsum",
          "nodeType": "host",
          "props": Object {
            "buttonStyle": undefined,
            "children": "Ipsum",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "Ipsum",
          "type": "Button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
